### DISCUSSION TITLE
[Feature] Option to manually remove video from live photos

### DISCUSSION THREAD

**Discussion by buxel** [immich-app/immich#4476] [Feature Request]
### The feature

I tend to use live photos to get a good shot but only occasionally really need/want the video part afterwards.

It would be great if there was a way to select a bunch of live photos and just remove the video part of it. In my case, this would reduce the required storage space by ~60%.

I understand that it may not seem worth the effort since storage is quite cheap but if this feature is easy enough to implement, I'd really appreciate it.

I had a short chat with Daniel on Discord [here](https://discord.com/channels/979116623879368755/994044917355663450).

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by Blendi** [comment]
I am also interested in this feature.
I tend to take a lot of pictures for work also. Most of them I don't need to keep the video, I need only the picture. When it comes to family pictures I really enjoy the live photos that our phones offer today. 
Having the option to delete/remove the video from live photos is a great feature that will make IMMICH feel more complete.

**Comment by maujogue** [comment]
Up for this feature. I recently uploaded a 10gigs google takeout, turns out half of it was videos from live photos I didn't know existed.

**Comment by kvalev** [comment]
That would be quite helpful, but given Immich's policy of never changing the original media I dont see it ever happening.

**Reply by maujogue** [reply]
It could create a duplicate, free of the live photo part. It would then be the user's responsibility to delete the original photo.

**Reply by je-kizer** [reply]
Immich allows you to delete photos and videos, I don't see how this would be different.

**Comment by bbrendon** [comment]
AI made you a script :D . I used it and seems to work great.  I would use the script to create a live photo album, then select all the images you want and put them in a new album. You can use the find param to get the UUID of an album for Live deletion.


https://pastebin.com/Y5JSAE36

**Reply by bbrendon** [reply]
Oh no!!! I noticed a lot of old live photos have two assets in Immich. The photo and the video are separate!! bugger.

**Comment by token47** [comment]
This now became a very needed feature since the new photo editing feature of 2.5+ do not work on live photos. I do not care about the video part so much after the initial shooting/adjusting, so I prefer to delete the video part so I can edit/rotate the photo.

**Reply by token47** [reply]
Even better, the video editor should offer to drop the video so it can edit it.

**Reply by YarosMallorca** [reply]
This will be addressed by #27251

**Comment by ptr727** [comment]
I'd love for a way to rid my library of the video portion of photos.\
I currently have a pre-import app i run to detect and delete the mov files.

If MOV is ~1 to 4s long\
Find movie photos pair; IMG_1234_HEVC.mov / IMG_1234.mov <-> IMG_1234.HEIC\
Use exiftools to get the `ContentIdentifier` tag for both files, if they match they are a set.\
Delete the MOV file.

**Comment by Sadashii** [comment]
++ Really need this considering the motion photos while a useful feature generally do tend to fill up storage, it would help a lot.

**Comment by adrianrudnik** [comment]
Had the same issue, added a [helper script](https://gist.github.com/adrianrudnik/8b7b7e2b34bdb758a8fdb6e67926f378) for this.

Edit: reworked it, due to the weird way immich handles those images. The script itself has a big intro description about what exactly happens, here is a gist:

```bash
# audit only, no changes
 IMMICH_URL="https://..." IMMICH_API_KEY="..." \
  ./immich-convert-live-photos-to-stills.sh \
  --host-library-path /var/lib/my-immich/data/library \
  --immich-library-path /data/library \
  --audit

# test on first image from the audit
mkdir ~/trash
 IMMICH_URL="https://..." IMMICH_API_KEY="..." \
  ./immich-convert-live-photos-to-stills.sh \
  --host-library-path /var/lib/my-immich/data/library \
  --immich-library-path /data/library \
  --move-to ~/trash \
  --convert-to-still \
  --limit 1
````

Remove limit to "clean" all, but **backup prior or use a clone to test against**, be very careful with it.

It is destructive, it unlinks the livePhotoVideoId from the original and trashes the related video, basically creating a still image, but moves a copy of the video to the `move-to` path. The script does not offer a restore-feature, so keep that in mind.

**Comment by hbilbo** [comment]
I think it would also be helpful to have an option to not sync MOVs at all when syncing photos on IOS. Removing live photos would be great for existing media, but IMO it would be better for future media to just not sync the live photo portion at all to immich. Unfortunately I haven't found a way to batch remove live photo data on IOS so it would have to be done on immich's side.

**Comment by OlliSchm** [comment]
I’d like to add a related use case from the mobile app. As Motion / Live Photos currently can’t be edited (crop, rotate, etc.), it would help a lot if the app could either:
* Strip the motion/video part from a Live/Motion Photo in place, or
* Create an editable still-image copy without the motion part, so I can keep that version and delete the original.

That way I can keep using Live/Motion Photos when shooting, then later turn a shot with an unneeded video into a normal photo that saves disk space and can be edited in Immich without re-uploading from another app.
This is one of the last remaining reasons I’d still have to bounce through Google Photos 👿 — being able to do this entirely in Immich would let me stay fully self-hosted.
